File: src/main/java/gmo/cashregister/services/admin/RightService.java
package gmo.cashregister.services.admin;

import gmo.cashregister.entity.security.Right;

import java.util.List;

public interface RightService {
    public List<Right> getRights();

    public Right getRight(long id);

    public List<Right> save(Right right);

    public List<Right> update(Right right);

    public List<Right> delete(long id);
}

File: front/cash-register-admin/src/services/RightService.js
import {RightRepository} from "../repositories/RightRepository";

const rightRepository = new RightRepository();

export class RightService {

    newRigth () {
        return rightRepository.newRight();
    }

    async getRights () {
        return await rightRepository.getRights();
    }

    async getRight (id) {
        return await rightRepository.getRight(id);
    }

    async save (right) {
        return await rightRepository.save(right);
    }

    async update (right) {
        return await rightRepository.update(right);
    }

    async delete (id) {
        return await rightRepository.delete(id);
    }

}